cmake/onnxruntime_rocm_hipify.cmake

```diff
@@ -114,6 +133,9 @@ if (NOT onnxruntime_USE_NCCL)
   list(APPEND contrib_ops_excluded_files "collective/distributed_reduce.cc")
   list(APPEND contrib_ops_excluded_files "collective/distributed_unsqueeze.cc")
   list(APPEND contrib_ops_excluded_files "collective/distributed_squeeze.cc")
+else()
```

Comment:
@tianleiwu this might be the critical fix instead of the glob wildcard.  Is there a reason the ROCm CI doesn't build with --enable_nccl and --use_mpi?

Reply:
https://github.com/microsoft/onnxruntime/blob/d5d9dbd51daf92e2535b190c27409c0e30640e6f/tools/ci_build/github/azure-pipelines/orttraining-pai-ci-pipeline.yml#L193-L201

I did not see those flags (--enable_nccl and --use_mpi) although there are `--nccl_home` and `--mpi_home`.

Reply:
Chose to open a new minimal PR that fixes the build failure but doesn't add --enable_nccl.

https://github.com/microsoft/onnxruntime/pull/19900